Have allsky.sh call variables.sh before starting capture

linuxkidd opened the pull request:
Allows all called scripts from `capture` to not need anything from `variables.sh`, thus the multi-loading guard in place in `variables.sh` will prevent unneeded extra work.

EricClaeys reviewed (commented):
Michael, I'm curious why you source's variables.sh near the end of the file rather than right before `source $ALLSKY_HOME/config.sh`, which is where I have it in my test version.  Having it higher means config.sh and most of allsky.sh can use the variables in variables.sh.

Also, before you merge this commit, I'd like to add some minor changes to allsky.sh.

EricClaeys commented:
Just looked - we really want the call to variables.sh above the call to config.sh since allsky.sh uses $ALLSKY_HOME/scripts that we need to replace with $ALLSKY_SCRIPTS.

linuxkidd commented:
@EricClaeys roger that... moved and updated all instances of `$ALLSKY_HOME/scripts` and `$ALLSKY_HOME/config` to use the appropriate variables instead.

EricClaeys reviewed (approved):
I'll add my changes after this PR is merged.
